diff --git a/ifopt-config.cmake b/ifopt-config.cmake
index ac24066..4bb15d4 100644
--- a/ifopt-config.cmake
+++ b/ifopt-config.cmake
@@ -44,18 +44,18 @@ include("${CMAKE_CURRENT_LIST_DIR}/ifopt_snopt-targets.cmake" OPTIONAL)
 get_target_property(ifopt_INCLUDE_DIRS ifopt::ifopt_core INTERFACE_INCLUDE_DIRECTORIES)
 
 # ifopt_core
-get_property(ifopt_LIB_CORE TARGET ifopt::ifopt_core PROPERTY LOCATION)
-list(APPEND ifopt_LIBRARIES ${ifopt_LIB_CORE})
+# get_property(ifopt_LIB_CORE TARGET ifopt::ifopt_core PROPERTY LOCATION)
+# list(APPEND ifopt_LIBRARIES ${ifopt_LIB_CORE})
 
 # ifopt_ipopt
-if(TARGET ifopt::ifopt_ipopt)
-  get_property(ifopt_LIB_IPOPT TARGET ifopt::ifopt_ipopt PROPERTY LOCATION)
-  list(APPEND ifopt_LIBRARIES ${ifopt_LIB_IPOPT})
-endif()
+# if(TARGET ifopt::ifopt_ipopt)
+#  get_property(ifopt_LIB_IPOPT TARGET ifopt::ifopt_ipopt PROPERTY LOCATION)
+#  list(APPEND ifopt_LIBRARIES ${ifopt_LIB_IPOPT})
+#endif()
 
 # ifopt_snopt
-if(TARGET ifopt::ifopt_snopt)
-  get_property(ifopt_LIB_SNOPT TARGET ifopt::ifopt_snopt PROPERTY LOCATION)
-  list(APPEND ifopt_LIBRARIES ${ifopt_LIB_SNOPT})
-endif()
+#if(TARGET ifopt::ifopt_snopt)
+#  get_property(ifopt_LIB_SNOPT TARGET ifopt::ifopt_snopt PROPERTY LOCATION)
+#  list(APPEND ifopt_LIBRARIES ${ifopt_LIB_SNOPT})
+#endif()
 
