diff -ruN tesseract-0.4.1/tesseract_urdf/cmake/tesseract_urdf-config.cmake.in tesseract/tesseract_urdf/cmake/tesseract_urdf-config.cmake.in
--- tesseract-0.4.1/tesseract_urdf/cmake/tesseract_urdf-config.cmake.in	2021-04-24 09:41:53.000000000 -0400
+++ tesseract/tesseract_urdf/cmake/tesseract_urdf-config.cmake.in	2021-05-28 23:14:26.588341260 -0400
@@ -12,7 +12,7 @@
 endif()
 find_dependency(Eigen3)
 find_dependency(console_bridge)
-find_dependency(PCL)
+find_dependency(PCL COMPONENTS common io)
 find_dependency(tesseract_geometry)
 find_dependency(tesseract_common)
 find_dependency(tesseract_scene_graph)
diff -ruN tesseract-0.4.1/tesseract_urdf/CMakeLists.txt tesseract/tesseract_urdf/CMakeLists.txt
--- tesseract-0.4.1/tesseract_urdf/CMakeLists.txt	2021-04-24 09:41:53.000000000 -0400
+++ tesseract/tesseract_urdf/CMakeLists.txt	2021-05-28 23:12:38.466040737 -0400
@@ -12,7 +12,7 @@
 find_package(Boost REQUIRED)
 find_package(Eigen3 REQUIRED)
 find_package(console_bridge REQUIRED)
-find_package(PCL REQUIRED)
+find_package(PCL REQUIRED COMPONENTS common io)
 find_package(tesseract_geometry REQUIRED)
 find_package(tesseract_common REQUIRED)
 find_package(tesseract_scene_graph REQUIRED)
