diff --git a/tesseract_python/cmake/__init__.py.in b/tesseract_python/cmake/__init__.py.in
index 6ddd63d..8a25717 100644
--- a/tesseract_python/cmake/__init__.py.in
+++ b/tesseract_python/cmake/__init__.py.in
@@ -1,3 +1,8 @@
 from __future__ import absolute_import
 
+import os
+try:
+    os.add_dll_directory(os.environ["CONDA_PREFIX"] + "/opt/tesseract_robotics/bin")
+except: pass
+
 from .@PY_MOD_NAME@ import *
\ No newline at end of file