diff --git a/tesseract_motion_planners/CMakeLists.txt b/tesseract_motion_planners/CMakeLists.txt
index 04289430..1fe86088 100644
--- a/tesseract_motion_planners/CMakeLists.txt
+++ b/tesseract_motion_planners/CMakeLists.txt
@@ -14,6 +14,7 @@ find_package(console_bridge REQUIRED)
 find_package(descartes_light REQUIRED)
 find_package(descartes_samplers REQUIRED)
 find_package(ompl REQUIRED)
+find_package(ode REQUIRED)
 find_package(tesseract_environment REQUIRED)
 find_package(tesseract_common REQUIRED)
 find_package(tesseract_command_language REQUIRED)
@@ -102,7 +103,7 @@ set(OMPL_SRC src/ompl/ompl_motion_planner.cpp
 
 message(AUTHOR_WARNING "OMPL INCLUDE DIRS: ${OMPL_INCLUDE_DIRS}")
 add_library(${PROJECT_NAME}_ompl ${OMPL_SRC})
-target_link_libraries(${PROJECT_NAME}_ompl PUBLIC ${PROJECT_NAME}_core ${OMPL_LIBRARIES} Boost::boost ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(${PROJECT_NAME}_ompl PUBLIC ${PROJECT_NAME}_core ${OMPL_LIBRARIES} ODE::ODE Boost::boost ${CMAKE_THREAD_LIBS_INIT})
 target_compile_options(${PROJECT_NAME}_ompl PRIVATE ${TESSERACT_COMPILE_OPTIONS_PRIVATE})
 target_compile_options(${PROJECT_NAME}_ompl PUBLIC ${TESSERACT_COMPILE_OPTIONS_PUBLIC})
 target_compile_definitions(${PROJECT_NAME}_ompl PUBLIC ${TESSERACT_COMPILE_DEFINITIONS})
diff --git a/tesseract_motion_planners/cmake/tesseract_motion_planners-config.cmake.in b/tesseract_motion_planners/cmake/tesseract_motion_planners-config.cmake.in
index 68f35e0f..3b457b63 100644
--- a/tesseract_motion_planners/cmake/tesseract_motion_planners-config.cmake.in
+++ b/tesseract_motion_planners/cmake/tesseract_motion_planners-config.cmake.in
@@ -6,6 +6,7 @@ set_and_check(@PROJECT_NAME@_LIBRARY_DIRS "${PACKAGE_PREFIX_DIR}/lib")
 
 include(CMakeFindDependencyMacro)
 find_dependency(ompl)
+find_dependency(ode)
 find_dependency(trajopt)
 find_dependency(trajopt_sco)
 find_dependency(descartes_light)
