diff --git a/tesseract_process_managers/src/core/process_planning_future.cpp b/tesseract_process_managers/src/core/process_planning_future.cpp
index 73f5ee3176..6624990480 100644
--- a/tesseract_process_managers/src/core/process_planning_future.cpp
+++ b/tesseract_process_managers/src/core/process_planning_future.cpp
@@ -27,6 +27,7 @@
 #include <tesseract_common/macros.h>
 TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
 #include <boost/serialization/nvp.hpp>
+#include <boost/serialization/map.hpp>
 #include <boost/serialization/unordered_map.hpp>
 #include <boost/serialization/vector.hpp>
 #include <boost/serialization/unique_ptr.hpp>
diff --git a/tesseract_process_managers/src/core/process_planning_problem.cpp b/tesseract_process_managers/src/core/process_planning_problem.cpp
index fcd1b8c001..f188414985 100644
--- a/tesseract_process_managers/src/core/process_planning_problem.cpp
+++ b/tesseract_process_managers/src/core/process_planning_problem.cpp
@@ -27,6 +27,7 @@
 #include <tesseract_common/macros.h>
 TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
 #include <boost/serialization/nvp.hpp>
+#include <boost/serialization/map.hpp>
 #include <boost/serialization/unordered_map.hpp>
 #include <boost/serialization/vector.hpp>
 #include <boost/serialization/unique_ptr.hpp>
diff --git a/tesseract_process_managers/src/core/process_planning_request.cpp b/tesseract_process_managers/src/core/process_planning_request.cpp
index b63291dfaf..c3086dbf86 100644
--- a/tesseract_process_managers/src/core/process_planning_request.cpp
+++ b/tesseract_process_managers/src/core/process_planning_request.cpp
@@ -26,6 +26,7 @@
 #include <tesseract_common/macros.h>
 TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
 #include <boost/serialization/nvp.hpp>
+#include <boost/serialization/map.hpp>
 #include <boost/serialization/unordered_map.hpp>
 #include <boost/serialization/vector.hpp>
 #include <boost/serialization/shared_ptr.hpp>
