diff --git a/trajopt/CMakeLists.txt b/trajopt/CMakeLists.txt
index d2dc08a..0e57a67 100644
--- a/trajopt/CMakeLists.txt
+++ b/trajopt/CMakeLists.txt
@@ -57,7 +57,7 @@ target_link_libraries(
          console_bridge::console_bridge
          Boost::boost
          Eigen3::Eigen
-         jsoncpp_lib)
+         JsonCpp::JsonCpp)
 target_compile_definitions(${PROJECT_NAME} PUBLIC ${TRAJOPT_COMPILE_DEFINITIONS})
 target_cxx_version(${PROJECT_NAME} PUBLIC VERSION ${TRAJOPT_CXX_VERSION})
 target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
diff --git a/trajopt/test/benchmarks/CMakeLists.txt b/trajopt/test/benchmarks/CMakeLists.txt
index b5207ba..3587755 100644
--- a/trajopt/test/benchmarks/CMakeLists.txt
+++ b/trajopt/test/benchmarks/CMakeLists.txt
@@ -11,7 +11,7 @@ macro(add_benchmark benchmark_name benchmark_file)
     ${benchmark_name}
     ${PROJECT_NAME}
     tesseract::tesseract
-    osqp::osqpstatic
+    osqp::osqp
     benchmark::benchmark)
   target_include_directories(${benchmark_name} PRIVATE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
   add_dependencies(${benchmark_name} ${PROJECT_NAME})
diff --git a/trajopt_sco/CMakeLists.txt b/trajopt_sco/CMakeLists.txt
index 488083a..57d8830 100644
--- a/trajopt_sco/CMakeLists.txt
+++ b/trajopt_sco/CMakeLists.txt
@@ -31,9 +31,10 @@ set(SCO_SOURCE_FILES
     src/modeling_utils.cpp
     src/num_diff.cpp)
 
-if(NOT APPLE AND NOT WIN32)
-  set(HAVE_BPMPD TRUE)
-endif()
+# if(NOT APPLE AND NOT WIN32)
+#   set(HAVE_BPMPD TRUE)
+# endif()
+set(HAVE_BPMPD FALSE)
 
 if(HAVE_BPMPD)
   add_executable(bpmpd_caller src/bpmpd_caller.cpp)
@@ -87,7 +88,7 @@ if(HAVE_BPMPD)
                                                      HAVE_BPMPD=ON)
 endif()
 if(osqp_FOUND)
-  target_link_libraries(${PROJECT_NAME} PRIVATE osqp::osqpstatic)
+  target_link_libraries(${PROJECT_NAME} PRIVATE osqp::osqp)
   target_compile_definitions(${PROJECT_NAME} PRIVATE HAVE_OSQP=ON)
 endif()
 if(qpOASES_FOUND)
@@ -102,7 +103,7 @@ target_link_libraries(
          Boost::boost
          Eigen3::Eigen
          ${CMAKE_DL_LIBS}
-         jsoncpp_lib)
+         JsonCpp::JsonCpp)
 target_compile_options(${PROJECT_NAME} PRIVATE ${TRAJOPT_COMPILE_OPTIONS_PRIVATE})
 target_compile_options(${PROJECT_NAME} PUBLIC ${TRAJOPT_COMPILE_OPTIONS_PUBLIC})
 target_compile_definitions(${PROJECT_NAME} PUBLIC ${TRAJOPT_COMPILE_DEFINITIONS})
diff --git a/trajopt_sco/test/CMakeLists.txt b/trajopt_sco/test/CMakeLists.txt
index 6affc26..901c640 100644
--- a/trajopt_sco/test/CMakeLists.txt
+++ b/trajopt_sco/test/CMakeLists.txt
@@ -43,7 +43,7 @@ target_link_libraries(
   GTest::Main
   ${PROJECT_NAME})
 if(osqp_FOUND)
-  target_link_libraries(${PROJECT_NAME}-test osqp::osqpstatic)
+  target_link_libraries(${PROJECT_NAME}-test osqp::osqp)
 endif()
 target_compile_options(${PROJECT_NAME}-test PRIVATE ${TRAJOPT_COMPILE_OPTIONS_PRIVATE}
                                                     ${TRAJOPT_COMPILE_OPTIONS_PUBLIC})
